### PR TITLE
chore(ui): add data-test to elements attribute

### PR DESCRIPTION
**Pull request by ltsuda**
This pull request adds attribute `data-testid` to web components to help test automation for developers that could use this project as a target, building it locally to not overload public website. The attribute `data-testid` is the default for `getByTestId` like selector in multiple Test Automation Frameworks like Microsoft Playwright, WebdriverIO and Testing Library.

Unit & Snapshot tests: PASSED
E2E local: PASSED

Closes: https://github.com/saucelabs/sample-app-web/issues/92

### CODE REVIEW

**Comment by ltsuda** [comment]
```bash
# npm test
Test Suites: 22 passed, 22 total
Tests:       129 passed, 129 total
Snapshots:   58 passed, 58 total
Time:        1.974 s, estimated 4 s
Ran all test suites.

# E2E
Spec Files:      8 passed, 8 total (100% completed) in 00:00:12 
```

File: src/components/Button.jsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ const Button = ({
       className={`btn${buttonTypeClass}${buttonSize}${extraClass}`}
       {...(testId
         ? {
-            "data-test": testId,
```
[the remark this answers is not in the document]
Comment (ltsuda) [reply src/components/Button.jsx:39]: I will wait for reply but I don't mind changing all to `data-test` as it seems to be some variations in these patterns on the internet.

[the remark this answers is not in the document]
Comment (diemol) [reply src/components/Button.jsx:39]: Then please change it to `data-test`.

[the remark this answers is not in the document]
Comment (ltsuda) [reply src/components/Button.jsx]: Renamed all `data-testid` to `data-test`

```bash

# npm test
Test Suites: 22 passed, 22 total
Tests:       129 passed, 129 total
Snapshots:   58 passed, 58 total
Time:        1.88 s, estimated 2 s
Ran all test suites.

# E2E locally
Spec Files:      8 passed, 8 total (100% completed) in 00:00:12

❯ chromedriver --version
ChromeDriver 110.0.5481.77 (65ed616c6e8ee3fe0ad64fe83796c020644d42af-refs/branch-heads/5481@{#839})

❯ google-chrome --version
Google Chrome 110.0.5481.30 dev

❯ node --version
v14.21.3

❯ date
Wed Jan 10 11:45:46 PM -03 2024
```

out

**Review by diemol — APPROVED** [review]
Thank you, @ltsuda

**Pull request merged**